packages/cache/src/internal/tar.ts

```diff
@@ -7,21 +7,18 @@ import {CompressionMethod} from './constants'
 
 const IS_WINDOWS = process.platform === 'win32'
 
-async function getTarPath(
-  args: string[],
-  compressionMethod: CompressionMethod
-): Promise<string> {
+async function getTarPath(args: string[]): Promise<string> {
+  let tarPath = await io.which('tar', true)
```

Comment:
I think this change does not look necessary now.

Reply:
Ah it may still be needed as L40 is also changed.

Reply:
Right. It was added to easy readability but now all if conditions are returning so let me remove that